src/test/testHooks.ts

```diff
@@ -0,0 +1,17 @@
+import { Context } from 'mocha';
+import { Telemetry } from '../platform/common/constants';
+import { sendTelemetryEvent } from '../telemetry';
+import { IS_CI_SERVER } from './ciConstants.node';
+
+export const rootHooks = {
+    afterEach(this: Context) {
+        let result = this.currentTest?.isFailed() ? 'failed' : this.currentTest?.isPassed() ? 'passed' : 'skipped';
```

Comment:
Can we add a condition `if (!IS_CI_SERVER) { return ; }` that way we don't send telemetry to server and we don't need to filter it. Else when ever we run any test locally we always end up sending that telemetry locally.